### PR TITLE
Add shared GitHub Actions workflow for ruby

### PR DESCRIPTION
**Pull request by archanaserver**
Used `theforeman/actions` reusable workflows for the GHA config.

https://github.com/theforeman/actions?tab=readme-ov-file#foreman-plugin-tests

### CODE REVIEW

**Comment by ekohl** [comment]
There, now it actually starts the action and we can see the result.

**Comment by archanaserver** [comment]
@ekohl They don't have rubocop, should we skip it for now, wdyt?

**Comment by ekohl** [comment]
Yes, you can skip it.

**Comment by evgeni** [comment]
Hmm. There are also no test tasks defined. "Cool"

**Comment by archanaserver** [comment]
How can I work in this @evgeni? and where also it is not defined :/

**Comment by archanaserver** [comment]
> Looking at https://ci.theforeman.org/job/test_plugin_matrix/5868/ruby=2.7,slave=fast/console it only calls `rake db:seed` to verify. But it's interesting that there is a test file. I think that hasn't ran in forever.

@ekohl would you mind expanding it a little more? I can't quite understand it.

File: .github/workflows/ruby_ci.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Ruby
+    uses: theforeman/actions/.github/workflows/foreman_plugin.yml@v0
+    with:
+      plugin: foreman_hooks
```
Comment (evgeni) [inline .github/workflows/ruby_ci.yml:18]:
```suggestion
      plugin: foreman_hooks
      matrix_exclude: '[{"task": "test:foreman_hooks"}]'
```

this should exclude the failing (non-existant) test and allow us to use the action for at least verifying the app boots with that engine.

Comment (archanaserver) [reply .github/workflows/ruby_ci.yml:18]: and it works

**Review by evgeni — APPROVED** [review]

**Pull request merged**